src/Controller/RedirectController.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\apigee_edge\Controller;
+
+use Drupal\Core\Entity\EntityBase;
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * Controller to redirect changed paths.
+ */
+class RedirectController extends ControllerBase {
+
+  /**
+   * The entity id from path input.
+   *
+   * @var string
+   */
+  private $entity_id;
+
+  /**
+   * Entity type from path input.
+   *
+   * @var string
+   */
+  private $entity_type_id;
+
+  /**
+   * Identity the entity id.
+   *
+   * @param mixed $entity_slug
+   *   Convert entity slug from path to id.
+   */
+  private function setEntityId($entity_slug) {
+    if (is_string($entity_slug)) {
+      $this->entity_id = $entity_slug;
+    }
+
+    elseif ($entity_slug instanceof EntityBase) {
+      $this->entity_id = $entity_slug->id();
+    }
+
+  }
+
+  /**
+   * Identity the type of entity.
+   *
+   * @param mixed $entity_slug
+   *   Entity slug from input.
+   */
+  private function setEntityTypeId($entity_slug) {
+    if ($entity_slug instanceof EntityBase) {
+      $this->entity_type_id = $entity_slug->getEntityTypeId();
+    }
+  }
+
+  /**
+   * Redirect the given path to the correct route.
+   *
+   * @param mixed $entity_slug
+   *   Entity slug from input.
+   * @param string $redirect_route
+   *   Route parameter defined as defined in routing.yml.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   Return a 404.
+   */
+  public function manageRedirect($entity_slug = NULL, $redirect_route = 'system.404') {
+
+    if ($entity_slug != NULL) {
+      $this->setEntityId($entity_slug);
+      $this->setEntityTypeId($entity_slug);
+      $redirect = $this->redirect($redirect_route, [$this->entity_type_id => $this->entity_id]);
+    }
+
+    else {
+      $redirect = $this->redirect($redirect_route);
+    }
+
+    return $redirect;
+
+  }
```

Comment:
```suggestion
  }

  /**
   * Grant access to "apigee_edge.redirect.developer_app.add_form_for_developer".
   *
   * @param mixed $entity_slug
   *   Entity slug from input.
   * @param string $redirect_route
   *   Route parameter defined as defined in routing.yml.
   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
   *   The route match.
   * @param \Drupal\Core\Session\AccountInterface $account
   *   The currently logged in account.
   *
   * @return \Drupal\Core\Access\AccessResultInterface
   *   The access result.
   */
  public function createDeveloperAppAccess($entity_slug, string $redirect_route, RouteMatchInterface $route_match, AccountInterface $account) {
    if ($entity_slug != NULL) {
      $this->setEntityId($entity_slug);
      $url = Url::fromRoute($redirect_route, ['user' => $this->entity_id]);
      return $url->access($account) ? AccessResult::allowed() : AccessResult::forbidden();
    }

    return AccessResult::forbidden('User is missing from route.');
  }
```

Reply:
I added another parameter because I thought I might need it for the teams redirects but I think those all can be handled by existing permissions. Let me know if you think I should change it back.

Reply:
I like the idea of making this more generic with the `$access_type`, and I think it will be needed, because the issue with having the legacy routes just check for permissions is that it ignores the team context and team permissions (they are different from drupal permissions). 
For example, `apigee_edge_teams.redirect.team.add_members` should be accessible to a user with the drupal permission `manage team members` or if they belong to the given team and have the team permission of `team_manage_members`. All of that logic is already taken care of in the access check of the redirect_route, so I think it would be easier to just construct the Url object of the redirect route and check access on it. LMK if it makes sense.

Reply:
Ah! That makes sense. Changes on the way.